src/main/java/com/google/sps/servlets/ServeImageServlet.java

```diff
@@ -0,0 +1,45 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+ 
+package com.googl.sps.servlets;
+ 
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+ 
+/**
+ * Serves the blob by passing the blob key to blobstore service 
+ * replaces the getServingUrl method (Obtains a URL that can dynamically 
+ * serve the image stored as a blob)
```

Comment:
This is all great stuff to document. I recommend splitting it up, though: Here (above the class), document What the class accomplishes (e.g. "Serves the image file corresponding to the given blob key"); and within the class, document How it works and Why (e.g. "This is a workaround because the standard approach using getServingUrl() isn't working").

Reply:
That's split up, thanks!